File: src/devng/adventofcode/day01/day01_test.py
#!/usr/bin/env python

from day01 import *
import unittest

class Day01Test(unittest.TestCase):
    def test_count_floor(self):
        f1 = count_floor("(())")
        self.assertEqual(f1, 0)

        f2 = count_floor("))(((((")
        self.assertEqual(f2, 3)

        f3 = count_floor(")())())")
        self.assertEqual(f3, -3)

    def test_find_basement(self):
        b1 = find_basement(")")
        self.assertEqual(b1, 1)

        b2 = find_basement("()())")
        self.assertEqual(b2, 5)

if __name__ == '__main__':
    unittest.main()

File: src/devng/adventofcode/day01/day01_test.go
package main

import "testing"

func TestCountFloor(t *testing.T) {
	c1 := CountFloor("(())")
	if c1 != 0 {
		t.Error("Wrong c1 value.")
	}

	c2 := CountFloor("))(((((")
	if c2 != 3 {
		t.Error("Wrong c2 value.")
	}

	c3 := CountFloor(")())())")
	if c3 != -3 {
		t.Error("Wrong c3 value.")
	}
}

func TestFindBasement(t *testing.T) {
	b1 := FindBasement(")")
	if b1 != 1 {
		t.Error("Wrong b1 value.")
	}

	b2 := FindBasement("()())")
	if b2 != 5 {
		t.Error("Wrong b2 value.")
	}
}
